seccomp/umount.go

```diff
@@ -117,15 +124,81 @@ func (u *umountSyscallInfo) process() (*sysResponse, error) {
 	return u.tracer.createContinueResponse(u.reqId), nil
 }
 
+// processUmmountAllowed purpose is to prevent immutable resources from being
+// unmounted.
+//
+// Method will return 'true' if the umount operation is deemed legit, and will
+// return 'false' otherwise.
+func (u *umountSyscallInfo) processUmountAllowed(
+	mip domain.MountInfoParserIface) (bool, *sysResponse) {
+
+	// There must be mountinfo state present for this target. Otherwise, let
+	// kernel handle the error.
+	info := mip.GetInfo(u.Target)
+	if info == nil {
+		return true, u.tracer.createContinueResponse(u.reqId)
+	}
+
+	// Return 'eperm' if the umount operation is over an immutable mountpoint.
+	if u.cntr.IsImmutableMountID(info.MountID) {
+		return false, u.tracer.createErrorResponse(u.reqId, syscall.EPERM)
```

Comment:
I am wondering if we should return EINVAL (target is not a mountpoint) instead of EPERM. The reason is that in the future we may hide immutable mounts inside the container, such that a process inside the container won't be able to know that these are mounts but will instead see them as regular files/directories. If that's the case, the EINVAL would be a more appropriate error.

Reply:
Also: this means we return an error for unmounts of /proc as well as the subdirs of /proc backed by sysbox-fs (e.g., /proc/sys). It's OK to return the error on the unmount of /proc itself, but for the subdirs we should ignore the unmount, as described in line 37 of this same file.

Reply:
Do we consider `/proc` and it's sysbox-fs backed sub-dirs (e.g., `/proc/sys`) as immutable here? if so, I have two comments:

1) We want to ignore unmounts subdirs of `/proc` backed by sysbox-fs. See line 37 of this same file for the rationale. This change would break this.

2) Why would we need the `if-else` in lines 65 and 117 of this file then? They would not be reachable it seems.

Reply:
Also: how come we are not checking for `(u.syscallCtx.root == "/")` here (see L79 for an example of when we use this check)? Don't we need to know that the process calling this has a root matching that of the sys container's init process?

Reply:
Got it.